python-backend/app/api/mines/mine/resources/mine_map.py

```diff
@@ -54,4 +46,30 @@ def get(self):
         response.last_modified = last_modified
         response.make_conditional(request)
 
-        return response
\ No newline at end of file
+        return response
+
+    @staticmethod
+    def rebuild_map_cache_async():
+        app = current_app._get_current_object()
+        environ = request.environ
+
+        def run_cache_rebuilding_thread():
+            with app.request_context(environ):
+                return MineMapResource.rebuild_and_return_map_cache()
+
+        thread = threading.Thread(target=run_cache_rebuilding_thread)
+        thread.start()
+
+    @staticmethod
+    def rebuild_and_return_map_cache():
+        records = MineMapViewLocation.query.filter(MineMapViewLocation.latitude != None).all()
+        last_modified = datetime.utcnow()
+
+        # jsonify then store in cache
+        map_result = json.dumps({
+            'mines': list((map(lambda x: x.json_for_map(), records)))
```

Comment:
Just an FYI, our `.json*` methods are being deprecated. I'm surprised this one still exists.

Reply:
ah, yeah I literally moved the existing code to a new method.  Is it worth fixing it with this pr, or should I work on that in a separate one?

Reply:
It's fine as-is, I think. But we should be aware of any time we're leveraging these json methods. If I'm seeing it correctly, this was already being used, so it's not a big deal here